src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,36 @@
-import React from 'react';
+import React, { ChangeEvent } from 'react';
+import { Status } from '../../types/Status';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { selectFilter, setQuery, setStatus } from '../../features';
+
+const statusOptions: Status[] = ['all', 'active', 'completed'];
 
 export const TodoFilter: React.FC = () => {
+  const { query, status } = useAppSelector(selectFilter);
+
+  const dispatch = useAppDispatch();
+
+  const setFilterOption = (event: ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setStatus(event.target.value as Status));
+  };
```

Comment:
The `useAppDispatch` hook should be called before any functions that use `dispatch`, such as `setFilterOption`. Consider moving the `dispatch` declaration above the `setFilterOption` function to ensure that `dispatch` is defined before it's used.